diff -up core/Common/base.pri.omv~ core/Common/base.pri
--- core/Common/base.pri.omv~	2023-07-30 19:46:17.023480040 +0200
+++ core/Common/base.pri	2023-07-30 19:46:46.419829883 +0200
@@ -227,11 +227,6 @@ core_linux {
     }
 }
 
-core_linux {
-    equals(TEMPLATE, app):CONFIG += core_static_link_libstd
-    plugin:CONFIG += core_static_link_libstd
-}
-
 core_win_32 {
     CORE_BUILDS_PLATFORM_PREFIX = win_32
 }
diff -up core/DesktopEditor/doctrenderer/js_internal/v8/inspector/inspector.pro.omv~ core/DesktopEditor/doctrenderer/js_internal/v8/inspector/inspector.pro
--- core/DesktopEditor/doctrenderer/js_internal/v8/inspector/inspector.pro.omv~	2023-07-30 19:47:02.772022658 +0200
+++ core/DesktopEditor/doctrenderer/js_internal/v8/inspector/inspector.pro	2023-07-30 19:47:04.756045961 +0200
@@ -7,8 +7,6 @@ CONFIG   -= app_bundle
 
 TEMPLATE = app
 
-CONFIG += core_static_link_libstd
-
 CORE_ROOT_DIR = $$PWD/../../../../../../core
 PWD_ROOT_DIR = $$PWD
 
diff -up core/DesktopEditor/doctrenderer/test/embed/test_embed.pro.omv~ core/DesktopEditor/doctrenderer/test/embed/test_embed.pro
--- core/DesktopEditor/doctrenderer/test/embed/test_embed.pro.omv~	2023-07-30 19:47:15.020166226 +0200
+++ core/DesktopEditor/doctrenderer/test/embed/test_embed.pro	2023-07-30 19:47:16.669185501 +0200
@@ -7,8 +7,6 @@ CONFIG   -= app_bundle
 
 TEMPLATE = app
 
-CONFIG += core_static_link_libstd
-
 CORE_ROOT_DIR = $$PWD/../../../../../core
 CORE_3DPARTY_DIR = $$CORE_ROOT_DIR/Common/3dParty
 PWD_ROOT_DIR = $$PWD
diff -up core/DesktopEditor/doctrenderer/test/internal/test_internal.pro.omv~ core/DesktopEditor/doctrenderer/test/internal/test_internal.pro
--- core/DesktopEditor/doctrenderer/test/internal/test_internal.pro.omv~	2023-07-30 19:47:34.789396507 +0200
+++ core/DesktopEditor/doctrenderer/test/internal/test_internal.pro	2023-07-30 19:47:36.622417770 +0200
@@ -7,8 +7,6 @@ CONFIG   -= app_bundle
 
 TEMPLATE = app
 
-CONFIG += core_static_link_libstd
-
 CORE_ROOT_DIR = $$PWD/../../../../../core
 PWD_ROOT_DIR = $$PWD
 
