diff -up desktop-apps/win-linux/ASCDocumentEditor.pro.p1~ desktop-apps/win-linux/ASCDocumentEditor.pro
--- desktop-apps/win-linux/ASCDocumentEditor.pro.p1~	2024-04-26 01:40:42.415108818 +0200
+++ desktop-apps/win-linux/ASCDocumentEditor.pro	2024-04-26 01:40:42.555110237 +0200
@@ -47,6 +47,9 @@ message($$PLATFORM_BUILD)
 updmodule:core_linux {
     SOURCES += $$PWD/res/gtk_resources.c
 }
+linux {
+    QT += x11extras
+}
 
 HEADERS += \
     src/prop/cmainwindowimpl.h
diff -up desktop-apps/win-linux/src/components/asctabwidget.h.p1~ desktop-apps/win-linux/src/components/asctabwidget.h
diff -up desktop-apps/win-linux/src/components/ctabpanel.h.p1~ desktop-apps/win-linux/src/components/ctabpanel.h
--- desktop-apps/win-linux/src/components/ctabpanel.h.p1~	2024-04-26 01:40:42.495109628 +0200
+++ desktop-apps/win-linux/src/components/ctabpanel.h	2024-04-26 01:40:42.555110237 +0200
@@ -4,7 +4,6 @@
 #include <QWidget>
 #include <QTimer>
 
-#include "qcefview.h"
 #include "casctabdata.h"
 
 class CTabPanel : public QWidget
