diff --git a/win-linux/defaults.pri b/win-linux/defaults.pri
index 4becf6d2..0e77e7c0 100644
--- a/win-linux/defaults.pri
+++ b/win-linux/defaults.pri
@@ -231,8 +231,12 @@ core_linux {
     PKGCONFIG += glib-2.0 gtk+-3.0 atk dbus-1 gtk+-unix-print-3.0 xcb
     LIBS += -lX11 -lX11-xcb
 
-    LIBS += $$CORE_3DPARTY_PATH/icu/$$PLATFORM_BUILD/build/libicuuc.so.58
-    LIBS += $$CORE_3DPARTY_PATH/icu/$$PLATFORM_BUILD/build/libicudata.so.58
+    use_bundled_icu {
+        LIBS += $$CORE_3DPARTY_PATH/icu/$$PLATFORM_BUILD/build/libicuuc.so.58
+        LIBS += $$CORE_3DPARTY_PATH/icu/$$PLATFORM_BUILD/build/libicudata.so.58
+    } else {
+        LIBS += -licuuc -licudata
+    }
 
     DEFINES += DOCUMENTSCORE_OPENSSL_SUPPORT
 }
