diff -up desktop-sdk/ChromiumBasedEditors/lib/src/cefview.cpp.omv~ desktop-sdk/ChromiumBasedEditors/lib/src/cefview.cpp
--- desktop-sdk/ChromiumBasedEditors/lib/src/cefview.cpp.omv~	2026-02-15 23:20:18.894007085 +0100
+++ desktop-sdk/ChromiumBasedEditors/lib/src/cefview.cpp	2026-02-15 23:20:50.563332752 +0100
@@ -387,7 +387,7 @@ public:
 		NSDirectory::CreateDirectory(m_sTempFolder);
 
 		m_pReader->SetTempDirectory(m_sTempFolder.c_str());
-		m_pReader->LoadFromFile(m_sFilePath, L"", m_sPassword, m_sPassword);
+		m_pReader->LoadFromFile(m_sFilePath, L"", m_sPassword.c_str(), m_sPassword.c_str());
 
 		switch (m_nFileType)
 		{
diff -up desktop-sdk/ChromiumBasedEditors/lib/src/fileprinter.cpp.omv~ desktop-sdk/ChromiumBasedEditors/lib/src/fileprinter.cpp
--- desktop-sdk/ChromiumBasedEditors/lib/src/fileprinter.cpp.omv~	2026-02-15 23:22:06.489418081 +0100
+++ desktop-sdk/ChromiumBasedEditors/lib/src/fileprinter.cpp	2026-02-15 23:22:48.799468731 +0100
@@ -970,7 +970,7 @@ void CCloudPDFSaver::GetResultPdf(const
 	CPdfFile oPdfResult(m_oPrintData.m_pApplicationFonts);
 	oPdfResult.SetTempDirectory(sTempDir);
 
-	oPdfResult.LoadFromFile(m_sPdfFileSrc, L"", m_sPdfFileSrcPassword, m_sPdfFileSrcPassword);
+	oPdfResult.LoadFromFile(m_sPdfFileSrc, L"", m_sPdfFileSrcPassword.c_str(), m_sPdfFileSrcPassword.c_str());
 	oPdfResult.EditPdf(sOutputFile);
 
 	CConvertFromBinParams oConvertParams;
@@ -1089,7 +1089,7 @@ DWORD CCloudPDFSaver::ThreadProc()
 			CPdfFile oPdfResult(m_oPrintData.m_pApplicationFonts);
 			oPdfResult.SetTempDirectory(sTempDir);
 
-			if (oPdfResult.LoadFromFile(sTmpFile, L"", m_sPdfFileSrcPassword, m_sPdfFileSrcPassword))
+			if (oPdfResult.LoadFromFile(sTmpFile, L"", m_sPdfFileSrcPassword.c_str(), m_sPdfFileSrcPassword.c_str()))
 			{
 				int nPagesCount = oPdfResult.GetPagesCount();
 				int nImageFormat = 4; // PNG
